src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/ThunkPool.cs

```diff
@@ -87,11 +87,7 @@ private unsafe ThunksHeap(IntPtr commonStubAddress)
 
             _allocatedBlocks = new AllocatedBlock();
 
-            IntPtr thunkStubsBlock;
-            lock (this)
-            {
```

Comment:
I was scratching my head about this - we're in a constructor. Nobody else can have `this`

Reply:
I'm not a fan of doing more within a PR than what's needed and this is orthogonal. Filed #116089.